src/lib/normalise-dag-node.js

```diff
@@ -33,65 +123,96 @@ export function normaliseDagPb (node, cid, type) {
   //   throw new Error('dag-pb multihash should match provided cid')
   // }
-  node = node.toJSON()
-  let format
+
+  const nodeAsJson = convertDagPbNodeToJson(node)
+  /**
+   * @type {NormalizedDagPbNodeFormat}
+   */
+  let format = 'unknown'
   try {
     // it's a unix system?
-    const { type, data, blockSizes } = UnixFS.unmarshal(node.data)
-    node.data = { type, data, blockSizes }
+    const { type, data, blockSizes } = UnixFS.unmarshal(nodeAsJson.data)
+    console.log('type, data, blockSizes: ', type, data, blockSizes)
+    nodeAsJson.data = { type, data, blockSizes }
     format = 'unixfs'
   } catch (err) {
     // dag-pb but not a unixfs.
     // console.log(err)
   }
+  const cidStr = toCidStrOrNull(cid)
+  if (cidStr == null) {
+    throw new Error(`cidStr is null for cid: ${cid}`)
+  }
 
   return {
-    cid,
+    cid: cidStr,
     type,
-    data: node.data,
-    links: normaliseDagPbLinks(node, cid),
-    size: node.size,
+    data: nodeAsJson.data,
+    links: normaliseDagPbLinks(nodeAsJson, cid),
+    size: nodeAsJson.size,
     format
   }
 }
 
-/*
+/**
  * Convert DagLink shape into normalized form that can be used interchangeably
  * with links found in dag-cbor
+ *
+ * @param {DagPbNodeAsJson} node
+ * @param {string} sourceCid
+ * @returns {NormalizedDagLink[]}
  */
 export function normaliseDagPbLinks (node, sourceCid) {
-  return node.links.map(({ name, size, cid }, index) => ({
-    path: name || `Links/${index}`,
-    source: sourceCid,
-    target: cid,
-    size,
-    index
-  }))
+  return node.links.map((link, index) => {
+    const { Name: name, Tsize: size, Hash: cid } = link
+
+    return ({
+      path: name || `Links/${index}`,
+      source: sourceCid,
+      target: cid.toString(),
+      size: size ?? 0,
+      index
+    })
+  })
 }
 
-/*
+/**
  * Find links and add type and cid info
+ *
+ * @function normaliseDagCbor
+ * @param {unknown} obj - The data object
+ * @param {string} cid - The string representation of the CID
+ * @param {number} code - multicodec code, see https://github.com/multiformats/multicodec/blob/master/table.csv
+ * @returns {NormalizedDagNode}
  */
-export function normaliseDagCbor (obj, cid, type) {
+export function normaliseDagCbor (obj, cid, code) {
   const links = findAndReplaceDagCborLinks(obj, cid)
   return {
     cid,
-    type,
+    type: code,
     data: obj,
-    links: links
+    links,
+    size: links.reduce((acc, { size }) => acc + size, 0),
+    format: 'unknown'
   }
 }
 
 /**
  * A valid IPLD link in a dag-cbor node is an object with single "/" property.
+ *
+ * @param {*} obj
+ * @param {*} sourceCid
+ * @param {*} path
+ * @returns {NormalizedDagLink[]}
  */
 export function findAndReplaceDagCborLinks (obj, sourceCid, path = '') {
-  if (!obj || !typeof obj === 'object' || Buffer.isBuffer(obj) || typeof obj === 'string') {
+  if (!obj || typeof obj !== 'object' || Buffer.isBuffer(obj) || typeof obj === 'string') {
```

Comment:
`!typeof obj === 'object'` always returns false because `(!typeof obj) === boolean` and `boolean === 'object'` is never true

<img width="581" alt="image" src="https://user-images.githubusercontent.com/1173416/178087568-cb2ea30c-6813-4426-96f6-352461d72126.png">

Reply:
yeah, that looks like a typo!